firebase-firestore/CHANGELOG.md

```diff
@@ -1,5 +1,9 @@
 # Unreleased
 
+# 18.0.1
+- [changed] Firestore streams now recovers more quickly from bad network states
```

Comment:
There’s a word agreement error here: either stream recovers or streams recover.

However, consider phrasing this without implementation details. Our users don’t know what “streams” are at all.

Something like, “Firestore client now recovers its connection more quickly after a network interruption.” Or something similar.

Reply:
I wanted to avoid "connection" since this PR doesn't actually affect that, whereas the PR (androidchannelbuilder) in 17.1.5 does. (See first entry in the changelog for that version.)

How about "Firestore client now recovers more quickly after a network interruption"?

Alternatively, do we need "client" in there? How about "Firestore now recovers more quickly after a network interruption"?

Reply:
I'm fine with leaving "connection" out but I also don't think it's misleading. The issue we have now is that we're not even attempting to retry our RPCs after resuming from airplane mode which delays the underlying library's attempts to connect. Similar to the way we want to leave implementation details out of the changelog it's OK to phrase things in terms that users understand even if they're not the best description in the strictest sense. (Again, not arguing necessarily for inclusion, just that I don't think it's harmful.)

Maybe we can describe this similarly to the 17.1.5 description but instead specify how this change is different? Maybe "Firestore now recovers more quickly from long periods without network access"?

Historically we've used "client" or "the SDK" in the changelog to refer to the SDK as an actor taking actions on behalf of user code.

Reply:
sg. I've applied your suggestion ("Firestore now recovers more quickly from long periods without network access").